pkg/ddevapp/config.go

```diff
@@ -747,6 +747,7 @@ type composeYAMLVars struct {
 	Hostnames                       []string
 	Timezone                        string
 	ComposerVersion                 string
+	DockerVersion                   string
```

Comment:
Let's use DockerEngineVersion here

Reply:
I decided to use a different technique: introduce a new custom function `templateCanUse(feature string)` for use in templates, which can perform more complex checks on the go side.